coordinator/testing/src/main/java/io/stargate/it/cql/compatibility/protocolv4/BatchStatementTest.java

```diff
@@ -0,0 +1,372 @@
+package io.stargate.it.cql.compatibility.protocolv4;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.config.OptionsMap;
+import com.datastax.oss.driver.api.core.config.TypedDriverOption;
+import com.datastax.oss.driver.api.core.cql.BatchStatement;
+import com.datastax.oss.driver.api.core.cql.BatchStatementBuilder;
+import com.datastax.oss.driver.api.core.cql.BoundStatement;
+import com.datastax.oss.driver.api.core.cql.DefaultBatchType;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.cql.QueryTrace;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.cql.Statement;
+import com.datastax.oss.driver.api.core.servererrors.InvalidQueryException;
+import io.stargate.it.BaseIntegrationTest;
+import io.stargate.it.driver.CqlSessionExtension;
+import io.stargate.it.driver.CqlSessionSpec;
+import io.stargate.it.storage.StargateConnectionInfo;
+import io.stargate.it.storage.StargateEnvironmentInfo;
+import java.util.Iterator;
+import java.util.List;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(CqlSessionExtension.class)
+@CqlSessionSpec(
+    initQueries = {
+      "CREATE TABLE test (k0 text, k1 int, v int, PRIMARY KEY (k0, k1))",
+      "CREATE TABLE counter1 (k0 text PRIMARY KEY, c counter)",
+      "CREATE TABLE counter2 (k0 text PRIMARY KEY, c counter)",
+      "CREATE TABLE counter3 (k0 text PRIMARY KEY, c counter)",
+    },
+    customOptions = "applyProtocolVersion")
+public class BatchStatementTest extends BaseIntegrationTest {
```

Comment:
hmm so this was missing before? batch test for the V4?

Reply:
OK I am kinda confused so all of those tests are for protocol v4 here? what about 5?

Reply:
Default tests should run for protocol v5, created these to confirm on compatibility with protocol v4.